components/Search.tsx

```diff
@@ -4,11 +4,15 @@ import SearchBar from "./SearchBar";
 import SearchResult from "./SearchResult";
 
 export default function Search() {
-  const [searchValue, setSearchValue] = React.useState<string>("");
+  const [searchValue, setSearchValue] = React.useState<string | null>(null);
 
   return (
     <>
-      <SearchBar onSearch={(value) => setSearchValue(value)} />
+      <SearchBar
+        searchValue={searchValue}
+        onInput={(value) => setSearchValue(value)}
```

Comment:
for naming event handlers, it's usually on-[a certain action/event]. It should make sense saying it out loud

```suggestion
        onSearch={(value) => setSearchValue(value)}
```